qiskit/primitives/backend_estimator_dev.py

```diff
@@ -0,0 +1,612 @@
+    places that work with :class:`~.BaseEstimator`, such as algorithms in
+    :mod:`qiskit.algorithms` including :class:`~.qiskit.algorithms.minimum_eigensolvers.VQE`.
+    However, if you're using a provider that has a native implementation of
+    :class:`~.BaseEstimator`, it is a better choice to leverage that native
+    implementation as it will likely include additional optimizations and be a
+    more efficient implementation. The generic nature of this class precludes
+    doing any provider- or backend-specific optimizations.
+    """
+
+    # pylint: disable=missing-raises-doc
+    def __init__(
+        self,
+        backend: Backend,
+        *,  # TODO: allow backend as positional after removing deprecations
+        abelian_grouping: bool = True,
+        skip_transpilation: bool = False,
+        bound_pass_manager: PassManager | None = None,
+        options: dict | None = None,
+    ) -> None:
+        """Initalize a new BackendEstimator instance.
+
+        Args:
+            backend: Required: The backend to run the primitive on.
+            abelian_grouping: Whether commuting observable components should be grouped.
+            skip_transpilation: If `True`, transpilation of the input circuits is skipped.
+            bound_pass_manager: An optional pass manager to run after parameter binding.
+            options: Default options.
+        """
+        # TODO: validation
+        self.backend = backend
+        self.abelian_grouping = abelian_grouping  # TODO: `group_commuting`
+        self.skip_transpilation = skip_transpilation  # TODO: tranpilation level
+        self._bound_pass_manager = bound_pass_manager  # TODO: standardize
+        super().__init__(
+            circuits=None,
+            observables=None,
+            parameters=None,
+            options=options,
+        )
+        self._transpile_options = Options()
+
+    def __getnewargs__(self) -> tuple:
+        return (self._backend,)
+
+    ################################################################################
+    ## PROPERTIES
+    ################################################################################
+    @property
+    def backend(self) -> BackendV2:
+        """Backend to use for circuit measurements."""
+        return self._backend
+
+    @backend.setter
+    def backend(self, backend: Backend) -> None:
+        if not isinstance(backend, Backend):
+            raise TypeError(
+                f"Expected `Backend` type for `backend`, got `{type(backend)}` instead."
+            )
+        # TODO: clear all transpilation caching
+        self._backend: BackendV2 = (
+            backend if isinstance(backend, BackendV2) else BackendV2Converter(backend)
+        )
+
+    @property
+    def abelian_grouping(self) -> bool:
+        """Groups commuting observable components."""
+        try:
+            return self._abelian_grouping
+        except AttributeError:
+            return True
```

Comment:
Why do you need to use `try-except`? Why don't you set the attribute in the constructor?
`try-except` in property makes the code complicated.

Reply:
It's a safeguard 🙂 I answered this here: https://github.com/Qiskit/qiskit-terra/pull/9074#discussion_r1028430395